CHANGELOG.md

```diff
@@ -35,6 +35,7 @@ Versioning](http://semver.org/spec/v2.0.0.html) except to the first release.
 - Decimal package use a test function GetNumberLength instead of a
   package-level function getNumberLength (#219)
 - Datetime location after encode + decode is unequal (#217)
+- queue.Take() returns an invalid task (#222)
```

Comment:
> So `nil` is an invalid task data? In which cases queue may return it then?

Before the patch it may happen:

task := queue.Take()
// task != nil
// task.Data() == nil
// task.Id() == 0
// task.Status() == ""

Reply:
It may heppen if MsgPack data was an array with one Nil element:
https://github.com/msgpack/msgpack/blob/master/spec.md#nil-format

Reply:
No, my question is more about "in what case queue would return `nil`". It seems that "Take was executed on RO instance" is the answer.

Reply:
> No, my question is more about "in what case queue would return `nil`". It seems that "Take was executed on RO instance" is the answer.

Oh, yes, exactly. But it seems to me this is a special case, others are possible.